### DISCUSSION TITLE
Add support to new kind,  cSRX

### DISCUSSION THREAD

**Discussion by urao** [srl-labs/containerlab#629] [Q&A] [answered]
Hi,
What is the procedure to add support to new kind, cSRX (https://www.juniper.net/documentation/product/us/en/csrx)

Thanks,
Umamaheshwar

**Comment by hellt** [comment]
Hey @urao 
adding containerized NOS such as cSRX should be relatively easy
basically, what one needs to do is to create a package for that NOS here https://github.com/srl-labs/containerlab/tree/master/nodes

you may want to base your work on the `cRPD` for example - https://github.com/srl-labs/containerlab/tree/master/nodes/crpd

then you just need to add this new kind to the list of supported kinds https://github.com/srl-labs/containerlab/blob/d92a8135f2b3d1e24c9004d2c6c5d5e3d6e64b56/nodes/node.go#L25-L47

**Comment by urao** [comment, marked as answer]
Thanks @hellt 
I can take cRPD as the base to add cSRX. To validate the changes, do you have a document/README on how to build the rpm/deb/binaries of containerlab?

**Reply by hellt** [reply]
to build just use `go build` and you will have a containerlab binary produced

to test the docs, I recommend using the following command:

```
docker run -it --rm -p 8000:8000 -v $(pwd):/docs squidfunk/mkdocs-material:7.1.8
```

This will start a local web server with live reload and you will be able to reach it at localhost:8000
